src/IonTextReader.ts

```diff
@@ -217,20 +217,19 @@ export class TextReader implements Reader {
     }
 
     annotations(): string[] {
-        let ann : string[] = [];
-        for (let str of this._parser.annotations()) {
-            ann.push(str);
-        }
-        for (let i = 0; i < ann.length; i++) {
-            if(ann[i].length > 1 && ann[i][0] === '$') {
-                let tempStr = ann[i].substr(1, ann[i].length);
-                if (+tempStr === +tempStr) {//look up sid, +str === +str is a one line is integer hack
-                    let symbol = this._symtab.getSymbolText(Number(tempStr));
-                    if(symbol === undefined || symbol === null) throw new Error("Unresolvable symbol ID, symboltokens unsupported.");
-                    ann[i] = symbol;
+        let ann: string[] = [];
```

Comment:
Rather than initializing a new array and populating it with `.forEach()`, I suggest using [`.map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

Reply:
huh, this is my first `map()` in TS/JS :)